fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/EnforceAndCostTask.java

```diff
@@ -324,8 +324,11 @@ private void setPropertyWithCost(GroupExpression groupExpression,
                                      PhysicalPropertySet outputProperty,
                                      List<PhysicalPropertySet> inputProperties) {
         groupExpression.setPropertyWithCost(outputProperty, inputProperties, curTotalCost);
+        // record the output/input properties when child group could satisfy this group expression required property
+        groupExpression.addValidOutputInputProperties(outputProperty, inputProperties);
```

Comment:
Should call this method only when isSetUseNthExecPlan is true

Reply:
done